magenta-lib/src/main/scala/magenta/tasks/FastlyTasks.scala

```diff
@@ -30,43 +31,73 @@ case class UpdateFastlyConfig(s3Package: S3Path)(implicit
   implicit val ec: ExecutionContextExecutorService =
     ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(10))
 
-  override def execute(
+  private def _execute(
+      client: FastlyApiClient,
       resources: DeploymentResources,
       stopFlag: => Boolean
-  ): Unit = {
-    FastlyApiClientProvider.get(keyRing).foreach { client =>
-      val activeVersionNumber =
+  ): Try[Int] = {
+    for {
+      activeVersionNumber <- Try(
         getActiveVersionNumber(client, resources.reporter, stopFlag)
-      val nextVersionNumber =
+      )
+      nextVersionNumber <- Try(
         clone(activeVersionNumber, client, resources.reporter, stopFlag)
-
-      deleteAllVclFilesFrom(
-        nextVersionNumber,
-        client,
-        resources.reporter,
-        stopFlag
       )
-
-      uploadNewVclFilesTo(
-        nextVersionNumber,
-        s3Package,
-        client,
-        resources.reporter,
-        stopFlag
+      _ <- Try(
```

Comment:
These might be slightly easier to read if the `Try` wrapper is added in the private methods rather than here

Reply:
I ended up removing the for comprehension and just wrapped the entire block in a `Try`. I think it achieves the same effect with fewer code changes =)

Reply:
Great, much less distracting to read as well